hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/MergeOnReadIncrementalRelation.scala

```diff
@@ -108,7 +110,9 @@ case class MergeOnReadIncrementalRelation(override val sqlContext: SQLContext,
         }.toSeq
       }
 
-      buildSplits(filterFileSlices(fileSlices, globPattern))
+      var filteredFileSlices = filterFileSlices(fileSlices, globPattern)
```

Comment:
let's undo the change in MOR incremental relation. we need to think through other corner cases.

Reply:
After reverting my changes the test is failing for the case when full table scan is enabled with incremental query.
`DataSourceReadOptions.INCREMENTAL_FALLBACK_TO_FULL_TABLE_SCAN_FOR_NON_EXISTING_FILES.key -> "true"`
So I have still kept the changes. We can think about it and revert if needed.

Reply:
But why is the test failing after revert? Does it fail even after setting `fileIndex.setIncludeLogFiles(false)`?

Reply:
yes. from what I understand, we wrote a new test and its failing. likely it will fail even w/ master code. So, lets de-couple and not block this patch for the test failure. lets file a follow up ticket and track it.